### ISSUE TITLE
Keyboard focus given to wrong view when coming out of "show desktop"

### ISSUE THREAD

**Issue by marcusbritanicus** [WayfireWM/wayfire#1405]
**Describe the bug**
While coming out of "Show desktop" (from wm-actions plugin), wayfire gives keyboard focus to the client which last held _**explicit keyboard focus**_ (see point 3, below), even if that client is on another workspace.

**To Reproduce**
This is quite easily reproducible - you'll need two apps (or one app with two windows) that take keyboard input.

1. Open a terminal/editor/any window that has a keyboard cursor (may not be needed specifically needed) in workspace 1.
2. Open another app/another window of the same app in workspace 2.
3. Go to workspace 1, click the input field in your window - this is what I meant above by explicit keyboard focus.
4. Switch to workspace 2. Now your second window will have cursor (keyboard) focus and will be the active view. Normally, if you start typing at this point, those keystrokes are relayed to this window no 2.
5. Activate "Show Desktop" by pressing the key-binding, and deactivate it by pressing the key binding again.

At this point, you'll notice that the window in the current workspace (workspace 2) is no longer set to active, and does not have keyboard focus. Type something, and what you typed will be sent to window 1 on workspace 1.

**Expected behavior**
View 2 on Workspace 2 (i.e. your current workspace) should be the active view, and should hold the keyboard focus.

**Wayfire version**
git

**Comment by soreau** [comment]
Is this still an issue with latest wayfire since e2efa57e?

**Comment by marcusbritanicus** [comment]
@soreau Thanks. Show desktop works as is expected. I updated wayfire to the latest git, and I do not see this issue.
